main/src/androidTest/java/cgeo/geocaching/activity/waypoint/AddWaypointActivityTest.java

```diff
@@ -15,42 +26,79 @@
 import static androidx.test.espresso.matcher.ViewMatchers.withId;
 import static androidx.test.espresso.matcher.ViewMatchers.withText;
 
+import java.util.Collections;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import static org.assertj.core.api.Java6Assertions.assertThat;
 import static org.hamcrest.Matchers.hasToString;
 import static org.hamcrest.Matchers.not;
 import static org.hamcrest.Matchers.startsWith;
 
-public class AddWaypointActivityTest extends AbstractAddWaypointActivityTest {
+@SuppressWarnings("PMD.JUnitTestsShouldIncludeAssert")
+public class AddWaypointActivityTest {
 
-    @Suppress
-    public static void testAddWayPointHasTypeSelection() {
-        onView(withId(R.id.type)).check(matches(isDisplayed()));
+    private Geocache testCache;
+
+    @Before
+    public void setUp() throws Exception {
+        testCache = CgeoTestUtils.createTestCache();
+        DataStore.saveCache(testCache, Collections.singleton(LoadFlags.SaveFlag.CACHE));
+
+//        super.setUp();
+//        setActivityIntent(new EditWaypointActivity_.IntentBuilder_(getInstrumentation().getContext()).geocode(getCache().getGeocode()).get());
+//        getActivity();
     }
 
-    // disabled because of issues with testing on Lollipop
+
+    @After
+    public void tearDown() {
+        CgeoTestUtils.removeCacheCompletely(testCache.getGeocode());
+    }
+
+    @Suppress // this will fail on CI since it does not have FINE-LOCATION granted
+    @Test
+    public void checkPermissions() {
+        assertThat(ContextCompat.checkSelfPermission(ApplicationProvider.getApplicationContext(), Manifest.permission.ACCESS_FINE_LOCATION))
+                .isEqualTo(PackageManager.PERMISSION_GRANTED);
+    }
```

Comment:
This test and more below are suppressed. Is it not possible to reactivate them?

Reply:
> This test and more below are suppressed. Is it not possible to reactivate them?

It is possible as soon as #13798 is resolved. This is why I don't want to delete the tests

Reply:
I have commented the tests accordingly